Add `show_conf` and `show_labels` for Solutions

RizwanMunawar opened the pull request:
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR enhances Ultralytics Solutions by improving label customization for object detection outputs and updating documentation for clarity and accuracy. 🏷️📚

### 📊 Key Changes
- **Unified Label Formatting:** Introduces a new `adjust_box_label` method to standardize how labels (class names, confidence scores, and track IDs) are displayed across all Solutions modules.
- **Configurable Display:** Adds options to show/hide labels and confidence scores via `show_labels` and `show_conf` settings.
- **Consistent Label Usage:** Updates all relevant modules (e.g., object counter, region counter, queue management, speed estimation) to use the new label formatting, ensuring consistent output.
- **Documentation Improvements:** Refines the export section in the tutorial notebook for clarity and updates the model export options table.
- **Author List Update:** Adds a new contributor to the documentation authors list.

### 🎯 Purpose & Impact
- **Improved Customization:** Users can now easily control whether labels and confidence scores appear on detection outputs, making results more tailored to their needs.
- **Cleaner Visuals:** Consistent and configurable labels lead to clearer, less cluttered visualizations, especially in tracking and counting scenarios.
- **Better Documentation:** Updated export documentation helps users understand model export options more easily.
- **Enhanced Collaboration:** Recognizing new contributors supports community engagement.

Overall, these changes make Ultralytics Solutions more flexible, user-friendly, and visually consistent! 🚀

RizwanMunawar committed:
add conf support

UltralyticsAssistant commented:
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline the review process. An Ultralytics engineer will also review your contribution soon.

Please ensure you have addressed the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Make sure your commit messages are clear and follow project conventions.
- ✅ **Synchronize with Source**: Confirm your PR is up to date with the `ultralytics/ultralytics` `main` branch. If not, update it using the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Check that all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. Address any failures as needed.
- ✅ **Update Documentation**: If your changes affect features or interfaces, please update the relevant [documentation](https://docs.ultralytics.com/).
- ✅ **Add Tests**: Where applicable, add or update tests to cover your changes, and verify that all tests pass.
- ✅ **Sign the CLA**: If this is your first Ultralytics PR, sign our [Contributor License Agreement (CLA)](https://docs.ultralytics.com/help/CLA/) by commenting "I have read the CLA Document and I sign the CLA" on this PR.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** needed for your fix or feature. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For additional details, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have any questions, feel free to leave a comment below.

Thank you for helping improve Ultralytics! 🌟🦾

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

UltralyticsAssistant committed:
Merge branch 'main' into conf-and-labels

RizwanMunawar committed:
Merge branch 'main' into conf-and-labels

RizwanMunawar committed:
add show_conf and show_labels support

RizwanMunawar committed:
Merge branch 'conf-and-labels' of https://github.com/ultralytics/ultralytics into conf-and-labels

RizwanMunawar committed:
Merge 2905801b114419a3649e1d153c562b2ecdcaec30 into 68ec1c1066c327ce04808050379ab7d0b1e1c2bd

UltralyticsAssistant committed:
Auto-update Ultralytics Docs Reference by https://ultralytics.com/actions

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

RizwanMunawar committed:
update solutions.py

RizwanMunawar commented:
@Laughing-q Hi, can you please review this PR?

It's based on user request [here](https://github.com/orgs/ultralytics/discussions/8112?sort=new#discussioncomment-12909967). Thanks

Laughing-q commented on ultralytics/solutions/object_counter.py:
```diff
@@ -1,5 +1,4 @@
 # Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license
-
```
@RizwanMunawar I think this line was removed by accident?

RizwanMunawar replied:
Oh really, don't know how :)

Laughing-q commented on ultralytics/solutions/speed_estimation.py:
```diff
@@ -83,7 +83,11 @@ def process(self, im0):
-            speed_label = f"{int(self.spd[track_id])} km/h" if track_id in self.spd else self.names[int(cls)]
+            speed_label = (
+                f"{int(self.spd[track_id])} km/h"
+                if track_id in self.spd and self.show_labels
```
I saw we already have such `show_labels` and `track_id` logic in `self.adjust_box_label`, why having it here?

RizwanMunawar replied:
@Laughing-q, yes, good catch, let me think about this!

RizwanMunawar committed:
Merge branch 'main' into conf-and-labels

RizwanMunawar committed:
fix object_counter.py line header space

RizwanMunawar committed:
Merge branch 'conf-and-labels' of https://github.com/ultralytics/ultralytics into conf-and-labels

RizwanMunawar committed:
fix object_counter.py line header space

RizwanMunawar committed:
update speed_estimation.py

RizwanMunawar committed:
update

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

RizwanMunawar commented:
@Laughing-q, ok, updated the code accordingly. Removed `self.spd = {}` from `SpeedEstimator`, it's now a part of `BaseSolution` class. But it will only be updated when the `SpeedEstimator` class is used. Also improved the `adjust_box_label` method. Let me know if any changes are required here. Thanks

Laughing-q commented:
@RizwanMunawar Oh I didn't notice there's a speed-estimator specific `spd` variable there(sorry), then probably the original implement would be fine. I would not recommend to having some subclass specific variable `spd` in superclass, which would make the super class bloated.

RizwanMunawar committed:
update speed_estimation.py

RizwanMunawar committed:
update

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

RizwanMunawar commented:
@Laughing-q all done! Good to merge?

glenn-jocher committed:
Merge branch 'main' into conf-and-labels

glenn-jocher committed:
Created using Colab

glenn-jocher committed:
Merge branch 'main' into conf-and-labels

Signed-off-by: Glenn Jocher <glenn.jocher@ultralytics.com>

RizwanMunawar committed:
Merge 19b84863396af463062e2add232ed91f88e60a88 into 4d9126ee010b84a665b6deccdf2f3f7e00d346d3

UltralyticsAssistant committed:
Auto-update Ultralytics Docs by https://ultralytics.com/actions

glenn-jocher committed:
Update tutorial.ipynb

Signed-off-by: Glenn Jocher <glenn.jocher@ultralytics.com>

UltralyticsAssistant commented:
Fantastic work, @RizwanMunawar, @Laughing-q, and @glenn-jocher! 🚀 This PR is now merged—thank you for elevating Ultralytics Solutions with improved label customization and clearer documentation. As Helen Keller said, “Alone we can do so little; together we can do so much.” Your collaborative efforts have made our tools more flexible, user-friendly, and visually consistent for everyone. We truly value your dedication and contributions to the Ultralytics community!
